### PR TITLE
✨ AMP Analytics: Added configuration and test files for RudderStack

### PR DESCRIPTION
**Pull request by dhawal-rudder**
Related i2i issue - [#28543](https://github.com/ampproject/amphtml/issues/28543)

### CODE REVIEW

**Comment by arnabp92** [comment]
@googlebot I signed it!

File: examples/analytics-vendors.amp.html [hunk 1]
```diff
@@ -1911,6 +1912,33 @@
   </script>
 </amp-analytics>
 <!-- end WebEngage Analytics example -->
+
+<!-- RudderStack example -->
+<amp-analytics type="rudderstack" id="rudderstack">
+  <script type="application/json">
+  {
+    "vars": {
+      "writeKey": "1XI9Niof0vuKt5XTLurbWh9KkXn",
```
Comment (micajuine-ho) [inline examples/analytics-vendors.amp.html:1921]: Nit: place example in alphabetical order.

File: examples/analytics-vendors.amp.html [hunk 2]
```diff
@@ -1911,6 +1912,33 @@
...
+        "on": "click",
+        "selector": "#test1",
+        "request": "track",
+        "vars": {
+          "eventName": "click"
+        }
+      }
+    }
+  }
+  </script>
```
Comment (micajuine-ho) [inline examples/analytics-vendors.amp.html:1939]: nit: correction indentation

File: extensions/amp-analytics/0.1/test/vendor-requests.json [hunk 3]
```diff
@@ -527,5 +527,10 @@
...
     "mediaPosition": "https://!trackDomain/!trackId/wt?p=440,_title_,1,_screen_width_x_screen_height_,_screen_color_depth_,1,,_document_referrer_,_viewport_width_x_viewport_height_,0&tz=_timezone_&eid=_client_id(amp-wt3-eid)_&la=_browser_language_&mi=!id&mk=pos&mt1=!currentTime&mt2=!duration&&pu=_source_url_&eor=1&x=!playedTotal",
     "mediaEnded": "https://!trackDomain/!trackId/wt?p=440,_title_,1,_screen_width_x_screen_height_,_screen_color_depth_,1,,_document_referrer_,_viewport_width_x_viewport_height_,0&tz=_timezone_&eid=_client_id(amp-wt3-eid)_&la=_browser_language_&mi=!id&mk=eof&mt1=!currentTime&mt2=!duration&&pu=_source_url_&eor=1&x=!playedTotal"
+  },
+  "rudderstack": {
+    "base": "?writeKey=!writeKey&context.library.name=Rudderstack AMP SDK&context.library.version=1.0.0&context.platform=AMP&anonymousId=_client_id(rl_anonymous_id)_&context.locale=_browser_language_&context.userAgent=_user_agent_&context.page.path=_canonical_path_&context.page.url=_canonical_url_&context.page.referrer=_document_referrer_&context.page.title=_title_&context.screen.width=_screen_width_&context.screen.height=_screen_height_",
```
Comment (micajuine-ho) [inline extensions/amp-analytics/0.1/test/vendor-requests.json:532]: Nit: place example in alphabetical order.

File: extensions/amp-analytics/0.1/vendors/rudderstack.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
+{
+  "transport": {
+    "beacon": false,
+    "xhrpost": false,
+    "image": true
```
Comment (micajuine-ho) [inline extensions/amp-analytics/0.1/vendors/rudderstack.json:5]: AMP-analytics will try beacon -> xhrpost -> image requests and handle the fallback. Just to confirm, your servers only accept image requests?

Comment (dhawal-rudder) [reply extensions/amp-analytics/0.1/vendors/rudderstack.json:5]: Yes, our servers only accept image requests.

**Comment by micajuine-ho** [comment]
@dhawal-rudder Have you signed the CLA?

**Comment by dhawal-rudder** [comment]
> @dhawal-rudder Have you signed the CLA?

We'll get it done in a day or two..will let you know once its done

**Comment by arnabp92** [comment]
@googlebot I signed it!

**Comment by dhawal-rudder** [comment]
@googlebot I signed it!

**Comment by dhawal-rudder** [comment]
@micajuine-ho Hey, we have signed the CLA.

**Pull request merged**